### PR TITLE
Flush the ChainIndexer when Coinview flushes

### PR DESCRIPTION
**Pull request by fassadlr**
On occasion when a user shuts down its node whilst syncing, the coin view tip gets set out of sync with the chain indexer/repository. This is because these 2 components have different flushing logic and as such we cant be guaranteed that they flush at the same time.

The solution here is to, when coinview flushes to disk, the chain indexer should also be persisted. This way they will never be out of sync (unless the user shuts down the node in between the 2 calls).

### CODE REVIEW

File: src/Stratis.StraxD/Properties/launchSettings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "profiles": {
     "Stratis.StraxD": {
-      "commandName": "Project"
```
Comment (quantumagi) [inline src/Stratis.StraxD/Properties/launchSettings.json:4]: This change can be reverted?

**Review by quantumagi — APPROVED** [review]
LGTM

**Review by zeptin — APPROVED** [review]

**Comment by noescape00** [comment]
`On occasion when a user shuts down its node whilst syncing, the coin view tip gets set out of sync with the chain indexer/repository.`

Yes. Same as with wallet when wallet can lag behind consensus. Why is it bad? 

Currently chain indexer takes maybe 2-3 times as long as it takes for rest of the node to sync, if we start waiting for chain indexer before advancing coinview we will slow down syncing dramatically. 

Also it opens a new attack vector: fill the blocks and then see node not being able to process a block in 16 seconds.

**Comment by fassadlr** [comment]
This has been resolved with a PR that now saves the full rewind history.

**Pull request closed without merging**